backend/backendserver/src/main/java/com/github/klucsik/birdnoiseserver/backendserver/services/DeviceLogService.java

```diff
@@ -0,0 +1,34 @@
+package com.github.klucsik.birdnoiseserver.backendserver.services;
+
+import com.github.klucsik.birdnoiseserver.backendserver.mappers.DeviceMapper;
+import com.github.klucsik.birdnoiseserver.backendserver.persistence.entity.DeviceLog;
+import com.github.klucsik.birdnoiseserver.backendserver.repository.DeviceLogRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class DeviceLogService {
+    private final DeviceLogRepository repository;
+    private final DeviceService deviceService;
+
+    public Long save(String chipId, String loglevel, String message) throws MethodArgumentNotValidException {
+        DeviceLog deviceLog = new DeviceLog();
+
+        deviceLog.setLogLevel(loglevel);
+        deviceLog.setMessage(message);
+        deviceLog.setDevice(DeviceMapper.MAPPER.Dtotodevice(deviceService.createUnregistered(chipId)));
```

Comment:
Here it always creates a device?

Reply:
We should rename the createUnregistered. I refactored it to be a FindByIdOrCreateNew, it will always give back a device(dto)